tckdb/backend/app/tests/endpoints/conftest.py

```diff
@@ -0,0 +1,214 @@
+# tckdb/backend/app/tests/conftest.py
+
+
+import os
+from pathlib import Path
+
+import pytest
+from alembic import command
+from alembic.config import Config
+from dotenv import load_dotenv
+from fastapi.testclient import TestClient
+from sqlalchemy import create_engine, text
+from sqlalchemy.orm import sessionmaker
+
+from tckdb.backend.app.db.query import SoftDeleteQuery
+from tckdb.backend.app.db.session import get_db
+from tckdb.backend.app.main import app
+from tckdb.backend.app.models.encorr import EnCorr
+from tckdb.backend.app.models.ess import ESS
+from tckdb.backend.app.models.level import Level
+
+# Load environment variables from .env.test
+# if os.getenv("RUNNING_IN_DOCKER") == "True":
+#     load_dotenv(dotenv_path="./app/core/.env.test.docker")
+# else:
+#     load_dotenv(dotenv_path="./tckdb/backend/app/core/.env.test")
+
+
+API_V1_STR = "/api/v1"
+POSTGRES_USER = os.getenv("POSTGRES_USER", "test_user")
+POSTGRES_PASSWORD = os.getenv("POSTGRES_PASSWORD", "test_pass")
+POSTGRES_DB = os.getenv("POSTGRES_DB", "test_db")
+POSTGRES_HOST = os.getenv("POSTGRES_HOST", "localhost")
+POSTGRES_PORT = os.getenv("POSTGRES_PORT", "5434")
+
+SQLALCHEMY_DATABASE_URL = f"postgresql://{POSTGRES_USER}:{POSTGRES_PASSWORD}@{POSTGRES_HOST}:{POSTGRES_PORT}/{POSTGRES_DB}"
+
+engine = create_engine(SQLALCHEMY_DATABASE_URL, echo=True)
```

Comment:
are we creating a session for each test file in separate? I think it will be better to have something like `session.test.py` that initialize a session and it can be imported by each test suit

Reply:
Yeah, the `db_session` and `client` fixtures have a scope="class", meaning they are instantiated once per test class. So each test class will receive its own client and session.

But yeah, we can centralize it so the tests receive the same session. Although, maybe we should discuss what the benefits are of this? At the moment though we only have the `batch` test, which will be doing `post` and `read` and `delete` in the same class so the session will be the same

Reply:
ok, we should check what are the best practices regarding sessions when testing with postgres and sqlalchemy. as long it does not affect our tests stability/run times im good with that

Reply:
not urgent for now